tensorflow/compiler/mlir/hlo/lib/Dialect/mhlo/transforms/legalize_tensor_load_op.cc

```diff
@@ -0,0 +1,103 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+// This file implements logic for lowering memref.tensor_load ops that are
+// inserted during `mhlo-legalize-to-lmhlo`.
+
+#include "mlir-hlo/Dialect/mhlo/transforms/passes.h"
+#include "mlir/Dialect/Shape/IR/Shape.h"
+#include "mlir/Dialect/StandardOps/IR/Ops.h"
+#include "mlir/Dialect/Tensor/IR/Tensor.h"  // TF:llvm-project
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/BuiltinOps.h"
+#include "mlir/IR/BuiltinTypes.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Pass/PassRegistry.h"
+#include "mlir/Support/LogicalResult.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+namespace mlir {
+namespace lmhlo {
+namespace {
+
+using shape::ShapeOfOp;
+using tensor::ExtractOp;
+
+struct LegalizeTensorLoadOpPass
+    : public mlir::PassWrapper<LegalizeTensorLoadOpPass, FunctionPass> {
+  void getDependentDialects(DialectRegistry& registry) const override {
+    registry.insert<shape::ShapeDialect, tensor::TensorDialect,
+                    memref::MemRefDialect>();
+  }
```

Comment:
This does not seem needed. You only need to list here the dialects that are introduced by the pass and *not matched in the input*

Reply:
Done